### PR TITLE
feat: error for parameter without schema or content

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Purpose:
- OAS3 schema requires that either a schema or a content object must be defined for any parameter. Tooling based on OAS3 will expect parameters to have a schema or content.

Changes:
- Add check that a schema is provided in content objects for parameters.
- Check that schema or content object provided for all parameters.

Test:
- Add test to ensure that error returned for parameter with no schema and no content objects.
- Add test to ensure a warning given for a content object in a parameter with no schema object.

Docs:
- Document the parameter-contains-schema rule.

### CODE REVIEW

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Review by mkistler — APPROVED** [review]
Looks good. 👍 

There's a small problem with the rule name.  Assuming you fix that no re-review needed.

File: docs/spectral-rules.md [hunk 1]
```diff
@@ -68,6 +68,32 @@ Validates that every path contains a path segment for the API major version, of
 
 **Default Severity**: warn
 
+## parameter-or-schema
```
[the remark this answers is not in the document]
Comment (barrett-schonefeld) [reply docs/spectral-rules.md]: Fixed documentation and rule name

**Pull request merged**